src/lib/bootloader/autoyast_converter.rb

```diff
@@ -0,0 +1,90 @@
+require "yast"
+
+module Bootloader
+  # Converter between internal configuration model and autoyast serialization of configuration.
+  class AutoyastConverter
+    class << self
+      include Yast::Logger
+
+      def import(_data)
+      end
+
+      def export(config)
+        log.info "exporting config #{config.inspect}"
+
+        bootloader_type = config.name
+        res = { "loader_type" => bootloader_type }
+
+        return res if bootloader_type == "none"
+
+        export_stage1(res, config.stage1) if config.respond_to?(:stage1)
+        export_default(res, config.grub_default)
+
+        res
+      end
+
+    private
+
+      STAGE1_MAPPING = {
+        "activate"      => :activate?,
+        "generic_mbr"   => :generic_mbr?,
+        "boot_root"     => :root_partition?,
+        "boot_boot"     => :boot_partition?,
+        "boot_mbr"      => :mbr?,
+        "boot_extended" => :extended_partition?
+      }
+      def export_stage1(res, stage1)
+        res["global"] ||= {}
```

Comment:
Well, you're modifying the argument you receive (`res`) and that's a side efect. `export_stage1` could return a new hash and the caller could be the responsible for merging. But, as I have said, I'm happy as long as it's documented.